mqemitter-mongodb.js

```diff
@@ -14,6 +15,54 @@ function toStream (op) {
   return op.stream ? op.stream() : op
 }
 
+function newId () {
+  return ObjectId.createFromTime(Date.now())
+}
+
+function connectClient (url, opts, cb) {
+  MongoClient.connect(url, opts)
+    .then(client => {
+      process.nextTick(cb, null, client)
+    })
+    .catch(err => {
+      process.nextTick(cb, err)
+    })
+}
+
+// check if the collection exists and is capped
+// if not create it
+async function checkCollection (ctx, next) {
+  // ping to see if database is connected
+  try {
+    await ctx._db.command({ ping: 1 })
+  } catch (err) {
+    ctx.status.emit('error', err)
+    return
+  }
+  const collectionName = ctx._opts.collection
+  const collections = await ctx._db.listCollections({ name: collectionName }).toArray()
+  if (collections.length > 0) {
+    ctx._collection = ctx._db.collection(collectionName)
+    if (!await ctx._collection.isCapped()) {
+      // the collection is not capped, make it so
+      await ctx._db.command({
+        convertToCapped: collectionName,
+        size: ctx._opts.size,
+        max: ctx._opts.max
+      })
+    }
+  } else {
+    // collection does not exist yet create it
+    await ctx._db.createCollection(collectionName, {
+      capped: true,
+      size: ctx._opts.size,
+      max: ctx._opts.max
+    })
+    ctx._collection = ctx._db.collection(collectionName)
+  }
+  process.nextTick(next)
```

Comment:
are you sure it shouldn't call `next` in the previous branch of the if?

Reply:
its just outside the if, so it always executes